website/search_migration/migrate.py

```diff
@@ -2,30 +2,46 @@
 # -*- coding: utf-8 -*-
 '''Migration script for Search-enabled Models.'''
 from __future__ import absolute_import
+
+import logging
 from modularodm.query.querydialect import DefaultQueryDialect as Q
+from scripts import utils as script_utils
 from website.models import Node
 from framework.auth import User
 import website.search.search as search
 
 from website.app import init_app
 
+logger = logging.getLogger(__name__)
+
 app = init_app("website.settings", set_backends=True, routes=True)
 
 
 def migrate_nodes():
-    for node in Node.find(Q('is_public', 'eq', True)
-            & Q('is_deleted', 'eq', False)):
+    n_iter = 0
+    nodes = Node.find(Q('is_public', 'eq', True) & Q('is_deleted', 'eq', False))
+    for node in nodes:
         node.update_search()
+        n_iter += 1
+
+    logger.info('Nodes migrated: {}'.format(n_iter))
 
 
 def migrate_users():
-    for user in User.find(Q('is_registered', 'eq', True)
-            & Q('date_confirmed', 'ne', None)):
-        user.update_search()
+    n_migr = 0
+    n_iter = 0
+    for user in User.find():
```

Comment:
So the ```is_active``` property does the following:
```python
(self.is_registered and
 self.password is not None and
 not self.is_merged and
 not self.is_disabled and
 self.is_confirmed())
```
```is_merged``` and ```is_disabled``` are properties, not fields, leaving ```is_registered```, ```password```, and ```date_confirmed``` (which is used in the ```is_confirmed``` function) as filterable fields.

Because we can't use properties in queries, the ```find_active``` function would be roughly equivalent to the original query in the migration script, with the addition of checking the password field. 

I am worried because the original query clearly did not find all active users, though looking at the functions it looks like it really should have (the original code should find more users than are active, not fewer). Do you know of any way to filter on the ```is_active``` property, rather than related fields, without iterating over all users?